pg_search/src/postgres/parallel.rs

```diff
@@ -78,6 +87,12 @@ pub unsafe fn maybe_init_parallel_scan(
         // ParallelWorkerNumber -1 is the main backend, which is where we'll set up
         // our shared memory information.  The mutex was already initialized, directly, in
         // `aminitparallelscan()`
+        //
+        // We always call init_without_mutex() here (even on rescans) because:
+        // 1. It properly resets remaining_segments for rescans
+        // 2. The mutex protects against race conditions - workers calling segments() will
+        //    acquire the same mutex and wait until initialization completes
```

Comment:
Not really related in this case, because this call is occurring under the mutex.
```suggestion
```

Reply:
Removed.